hack/bats/tests/preserve-env.bats

```diff
@@ -135,16 +125,93 @@ local_setup() {
     assert_line FOO=foo
     assert_line FOOBAR=foobar
     assert_line BAR=bar
-    refute_line --regexp '^BARBAZ='
+    assert_line BARBAZ=barbaz
 }
 
-@test 'setting both allow list and block list generates a warning' {
-    export LIMA_SHELLENV_ALLOW=FOO
-    export LIMA_SHELLENV_BLOCK=BAR
+@test 'wildcard patterns work in all positions and combinations' {
+    # Test wildcard at middle, and other combinations
+    export LIMA_SHELLENV_BLOCK="FOO*BAR,*FOO*BAR*,*TEST*,*SUFFIX"
```

Comment:
There is no point in adding `FOO*BAR` because everything it matches is also matched by `*FOO*BAR*`.

I think to keep it simple, this should be the only pattern used in this test.